idea/src/org/jetbrains/kotlin/idea/inspections/conventionNameCalls/ReplaceCallWithBinaryOperatorInspection.kt

```diff
@@ -148,4 +149,19 @@ class ReplaceCallWithBinaryOperatorInspection : AbstractApplicabilityBasedInspec
             else -> OperatorConventions.BINARY_OPERATION_NAMES.inverse()[identifier]
         }
     }
+
+    private fun KtDotQualifiedExpression.isFloatingPointNumberEquals(): Boolean {
```

Comment:
This implementation is very strange. You focus here on a given example, but there are some simpler cases which probably will not be detected by this implementation correctly. I'd suggest you should find (via `resolveToCall()`) type of `equals` receiver and type of `equals` argument, and in case both are floating-point types, the function should return `true`, otherwise `false`.

Reply:
Thanks for your comments and suggestions. It's fixed now. How can I handle the following cases?

```kotlin
fun test(a: Any, b: Any) =
        a is Double && b is Double &&
                a.equals(b) // Here a and b are statically known to be Double.
```

Reply:
Looks like it's a hard question and a hard task. I will implement this piece myself